x-pack/plugins/alerting/server/application/rule/methods/get_schedule_frequency/get_schedule_frequency.ts

```diff
@@ -0,0 +1,115 @@
+  };
+}
+
+const convertIntervalToFrequency = (context: RulesClientContext, schedule: string) => {
+  let scheduleFrequency = 0;
+
+  try {
+    // Normalize the interval (period) in terms of minutes
+    scheduleFrequency = convertDurationToFrequency(schedule);
+  } catch (e) {
+    context.logger.warn(
+      `Failed to parse rule schedule interval for schedule frequency calculation: ${e.message}`
+    );
+  }
+
+  return scheduleFrequency;
+};
+
+export const getScheduleFrequency = async (
+  context: RulesClientContext
+): Promise<GetScheduleFrequencyResult> => {
+  const response = await context.internalSavedObjectsRepository.find<
+    RuleDomain,
+    SchedulesIntervalAggregationResult
+  >({
+    type: 'alert',
+    filter: 'alert.attributes.enabled: true',
+    namespaces: ['*'],
+    aggs: {
+      schedule_intervals: {
+        terms: {
+          field: 'alert.attributes.schedule.interval',
+        },
+      },
+    },
+  });
+
+  const buckets = response.aggregations?.schedule_intervals.buckets ?? [];
+
+  const totalScheduledPerMinute = buckets.reduce((result, { key, doc_count: occurrence }) => {
+    const scheduleFrequency = convertIntervalToFrequency(context, key);
+
+    // Sum up all of the frequencies, since this is an aggregation.
+    return result + scheduleFrequency * occurrence;
+  }, 0);
+
+  const result = {
+    totalScheduledPerMinute,
+    remainingSchedulesPerMinute: Math.max(
+      context.maxScheduledPerMinute - totalScheduledPerMinute,
+      0
+    ),
+  };
+
+  try {
+    getSchemaFrequencyResultSchema.validate(result);
+  } catch (e) {
+    context.logger.warn(`Error validating rule schedules per minute: ${e}`);
+  }
+
+  return result;
+};
+
+interface ValidateScheduleLimitParams {
+  context: RulesClientContext;
+  prevInterval?: string | string[];
+  updatedInterval: string | string[];
+}
+
+export const validateScheduleLimit = async (params: ValidateScheduleLimitParams) => {
+  const { context, prevInterval = [], updatedInterval = [] } = params;
+
+  const prevIntervalArray = Array.isArray(prevInterval) ? prevInterval : [prevInterval];
+  const updatedIntervalArray = Array.isArray(updatedInterval) ? updatedInterval : [updatedInterval];
+
+  const prevSchedulePerMinute = prevIntervalArray.reduce((result, interval) => {
+    const scheduleFrequency = convertIntervalToFrequency(context, interval);
+    return result + scheduleFrequency;
+  }, 0);
+
+  const updatedSchedulesPerMinute = updatedIntervalArray.reduce((result, interval) => {
+    const scheduleFrequency = convertIntervalToFrequency(context, interval);
+    return result + scheduleFrequency;
+  }, 0);
+
+  const { remainingSchedulesPerMinute } = await getScheduleFrequency(context);
+
+  // Compute the new remaining schedules per minute if we are editing rules.
+  // So we add back the edited schedules, since we assume those are being edited.
+  const computedRemainingSchedulesPerMinute = remainingSchedulesPerMinute + prevSchedulePerMinute;
```

Comment:
Should we ignore the rule that we are trying to update in the calculation of `totalScheduledPerMinute`

Reply:
Ignore me here, I did not understand it at first but we are doing the right thing.